spec/models/publisher_spec.rb

```diff
@@ -87,3 +87,104 @@
               publish
   end
 end
+
+RSpec.describe Publisher, "#discard_draft" do
+  let :publishing_api do
+    double(:publishing_api)
+  end
+
+  let(:publishing_api_which_always_fails) do
+    api = double(:publishing_api)
+    gds_api_exception = GdsApi::HTTPErrorResponse.new(422,
+                                                      'https://some-service.gov.uk',
+                                                      {'error' => {'message' => 'trouble'}})
+    allow(api).to receive(:discard_draft).and_raise(gds_api_exception)
+    api
+  end
+
+  let :subject do
+    Publisher.new(content_model: guide, publishing_api: publishing_api)
+  end
+
+  before do
+    allow(publishing_api).to receive(:discard_draft)
+  end
```

Comment:
Could we move the stubbing of this method to the let definition? It just means less jumping around the file for a reader when they drop in to read a single test.

And just to voice a personal preference.. I'm not super pumped for subject blocks either. I constantly forget what the subject is. Is it a publishing api that always passes or a publishing api that always fails? My preference in this case would be a..

```
let :publishing_api_which_always_succeeds do
  publishing_api = double(:publishing_api)
  allow(publishing_api).to receive(:discard_draft)
  publishing_api
end
```
and then just used in the example..
```
it do
  publisher = Publisher.new(content_model: some_guide, publishing_api: publishing_api_that_always_succeeds)
  expect(publisher.discard_draft).to be_success
end
```
You end up with a bit of repetition but I think it's more readable for future generations.